### PR TITLE
leader election needs to operate configmaps and leases

### PR DESCRIPTION
**Pull request by haoqing0110**
Related PR: https://github.com/open-cluster-management-io/registration-operator/pull/260
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: deploy/spoke/role.yaml [hunk 1]
```diff
@@ -6,6 +6,16 @@ metadata:
...
 rules:
+# leader election needs to operate configmaps and leases
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["create", "get", "list", "update", "watch", "patch"]
+# create hub-kubeconfig external-managed-registration/work secrets
+# TODO(zhujian7): may be replaced by a clusterrole to grant secret operation for others namespaces when addon
```
Comment (qiujian16) [inline deploy/spoke/role.yaml]: this TODO is not needed, right?

Comment (haoqing0110) [reply deploy/spoke/role.yaml]: yes, removed that.

**Pull request merged**